src/sys/options.cxx

```diff
@@ -109,6 +137,53 @@ const Options &Options::operator[](const std::string &name) const {
   return it->second;
 }
 
+std::multiset<Options::FuzzyMatch>
+Options::fuzzyFind(const std::string& name, std::string::size_type distance) const {
```

Comment:
There are various choices we could make about how this actually functions. I've chosen a default max distance of 4, so an option with more than 4 edits from the search string aren't counted, for example: `some_string` won't match `another_string` (by default).

It also matches an "unqualified name" (e.g. `first`) against a "fully qualified" name (e.g. `mesh:ddx:first`) with a distance of 1. But it won't try to strip the parent sections and match that. So `mesh:ddx:first` won't match `some:other:first`.

Another check is matching different cases: `THIS` matches `this` matches `ThIs`, all distance 1.

These choices were a little arbitrary, so definitely happy to hear suggestions on other choices